lib/fetch/index.js

```diff
@@ -1759,7 +1759,12 @@ async function httpNetworkFetch (
       // 1. If connection uses HTTP/2, then transmit an RST_STREAM frame.
       fetchParams.controller.connection.destroy()
 
-      // 2. Return the appropriate network error for fetchParams.
+      // 2. If error is ProxyAgent related, then pass it.
```

Comment:
this comment isn't related to the fetch spec, the comment changes should be reverted. Instead, add a comment explaining why this is needed or link to the issue.